Autopilot/PathManager/Src/landingTransitionStage.cpp

```diff
@@ -0,0 +1,65 @@
+#include "pathStateClasses.hpp"
+
+void landingTransitionStage::execute(pathManager* pathMgr)
+{
+    if(!pathMgr -> madeLandingPoints)
+    {
+        //requires data structure that dhruv wants to use 
+        path = LandingManager::createSlopeWaypoints(getFromTelemetry::telemetryInput);
+
+        //creating waypoints 
+        pathArray[0] = landingPath.initialize_waypoint(path.intersectionPoint.longitude, path.intersectionPoint.latitude, path.intersectionPoint.altitude, PATH_FOLLOW);
+        pathArray[1] = landingPath.initialize_waypoint(path.aimingPoint.longitude, path.aimingPoint.latitude, path.aimingPoint.altitude, PATH_FOLLOW);
+        pathArray[2] = landingPath.initialize_waypoint(path.stoppingPoint.longitude, path.stoppingPoint.latitude, path.stoppingPoint.altitude, PATH_FOLLOW);
+       
+        //initializing flight path, QUESTION!!! what should I do with currentPosition parameter? its a PathData struct, but it requires info from sensorfusion
+        waypointStatus = landingPath.initialize_flight_path(pathArray, 3,      );
+        
+        //set made madelandingPoints to true
+        pathMgr->madeLandingPoints = true;
+    }
+
+    //follow the landing waypoints
+    waypointStatus = landingPath.get_next_directions(getFromTelemetry::telemetryInput, &cruisingState::_outputdata);
+
+    //calculating the difference in heading to detect if finished turning (2 differences in heading possible)
+    differenceInHeading1 = getFromTelemetry::telemetryInput.landingDirection - sensorFusion::input.track;
+    differenceInHeading2 = sensorFusion::input.track - getFromTelemetry::telemetryInput.landingDirection;
```

Comment:
The reason I'm doing this is to ensure tracks that are around 360 are accounted for. So for example, the landing direction is 355 and the current track is 10 degrees. I would get a differenceInHeading1 = 345, and the differenceInHeading2 =-345. Then, I add 360 to the directions if they are negative, so differenceInHeading1 = 345 and the differenceInHeading2 = 15. This results in the smaller heading being the difference we want. In this example, that would be 15 degrees rather than 345 degrees in the other direction.